### DISCUSSION TITLE
0.0.50 tests on S42B and S57B

### DISCUSSION THREAD

**Discussion by jlmxyz** [CAP1Sup/Intellistep#115] [General]
Hi,
today I had time to test on S42B and S57B a version compiled from 37e736f1cc814663912333bc32fba36650426ef6 commit
since both boards seems similar, I wanted to check if what I saw on one was due to difference of board or not....

first I was impressed by the ease of flashing and even enter debug! I remember old times where jtags setup took more than a day just to establish communication, here it's really easy, just enter the pio plugin, upload and that's it!

S57B :

1. see boot screen : ok
2. see calibration status sceen : ok, but if calibration data is saved to flash, I can't enter the menu again... pressing any pressbutton except reset has no effect...
3. calibration : seems that calibration don't make the motor move like others calibrations on closed loop controlers???
4. most of the time, the status screen isn't displayed.... looks like a crash? 
5. when the status screen is displayed, I could make the motor move
             1. most of the time the motor was moving the carriage about 1cm back very fast and 1cm forward at moderate speed
             2. once I could make it only move in the requested direction for 1cm when asking for +X mouve and 1cm in other direction
             3. after a few move the status display seems to freeze, and no move can be performed
             4. noise is correct 
             5. never managed to get any communication on serial line at 115200-8n1 settings, tried with RxTx or TxRx on TxRx pins, according to what I saw in the code, there should be an echo... but nothing, could be really great to have debug info printed on the serial...
             6. after a few second (<10s) the display seems to freeze.... and no communication, I didn't saw also the counter status increased on incomming steps...

S42B : 

1. never managed to make the motor move.... BUT
2. I never saved calibration result to flash, push button works fine, could enter the menu several time, change the power settings but the motor didn't moved... once I got a 1degree move but never could reproduce
3. on status screen I see a step increase and decrease of the count that I sent, so it seems that STP-DIR-EN interface is working fine
4. same, serial don't echo anything...

so seems that on S57B there is an instability that crash the firmware.... but that the firmware can make the motor move.... and on S42B the firmware seems not to crash but motor don't move....

best regards

**Comment by jlmxyz** [comment]
made further testing on S57B : 
seems that the hang (or black screen) between the calibration boot screen and the status screen occurs when there is an on going communication, I had to reset the board after issuing a 20cm move, and after each reboot the screen gone black and the carriage moved a little.... so maybe a bug at init if there is some ongoing communication...

**Reply by CAP1Sup** [reply]
Check out PR #113 . I think that the issue you're seeing can be resolved by reducing the `SYSCLK_FREQ` from 128 to 72. Try that, let me know how it goes. Print results would also be greatly appreciated. Unfortunately, I'm too busy at the moment to be able to do the testing myself.

**Reply by jlmxyz** [reply]
ok setting SYSCLK_FREQ to 72 do solve the crash issue
but I still get some strange behaviour... each movement begins by a backward movement of 1cm about
I need to do more testing with +-500+ mm  movements, because it seems to move every time in the same direction (display counter seems to behave correctly, I have to check about microsteps settings on control board and motor)

what I see so far  after a +-500mm move : 
- a fast move backward (noise ok)
- a slow move in (same?) direction (awful noise, lot of vibrations, as if the motor was micro stepping forward and backward to stabilize)
- if I issue another move command I managed to perform very fast move, which were REALLY REALLY QUIET

to give a "perceptual noise level" 
- move really fast : about 30db
- move slow : about 40db
- move fast :  about 32db

serial is working (but no echo, so it's missleading since GCodes need to be <> enclosed) but needs rework from what I see in code (stop main loop during communication, this should be performed during interrupt and flush of buffer when stop sequence is found, main loop should only check for full commands)

I'll add some GCodes (move and settings) so that 
- movement can be controlled using serial (can be useful for debug) 
- have the menu of oled displayed on serial
- disable/enable dip switch use

and complete oled display to have the settings tunable from serial be available on display (for exemple microsteppings, more values are available on serial than on display, and motor stepping can't be set through display (0.9 vs 1.8) )

same will add some GCode to set board freq... so that default will be 72, can be set to 128 but if thermal protection rise, go back to 72 as a fail safe...

best regards

**Reply by CAP1Sup** [reply]
Sounds good! Let me know how it turns out!
